deployment/deployutils/wizardInputs.cpp

```diff
@@ -86,13 +86,26 @@ void CWizardInputs::setEnvironment()
   if(m_pXml->hasProp("@ipList"))
     formIPList(m_pXml->queryProp("@ipList"), m_ipaddress);
 
+  if(m_pXml->hasProp("@roxieNodes"))
+    m_roxieNodes = atoi(m_pXml->queryProp("@roxieNodes"));
+
```

Comment:
I don't see where m_roxieNodes is initialized. I looked in the header, but perhaps I missed it. Same with m_thorNodes below.

Reply:
envgen call deployutils.cpp function:
buildEnvFromWizard(optionsXml, pServiceName, pCfg, envXml, arrBuildSetWithAssignedIPs, arrAssignIPRanges, &dirMap);

optionalXml has supportNodes, roxieNodes, espNode and thorNodes variables
optionsXml.appendf("<XmlArgs supportNodes=\"%d\" roxieNodes=\"%d\" thorNodes=\"%d\" espNodes=\"%d\" slavesPerNode=\"%d\" thorChannelsPerSlave=\"%d\" roxieChannelsPerSlave=\"%d\" roxieOnDemand=\"%s\" ipList=\"%s\"/>", supportNodes, roxieNodes,
      thorNodes, espNodes, slavesPerNode, thorChannelsPerSlave, roxieChannelsPerSlave, roxieOnDemand?"true":"false", ipAddrs.str());

buildEnvFromWizard create CWizardInputs instance with all nodes numbers assigned to instance variable m_xxx

CWizardInputs wizardInputs(wizardXml, service, cfg, dirMap, arrBuildSetWithAssignedIPs, arrAssignedIPs);

Reply:
@xwang2713 I will approve, but I still think you should initialize these values in the constructor initialization list.